graylog2-web-interface/src/views/components/MigrateFieldCharts.jsx

```diff
@@ -0,0 +1,207 @@
+import LineVisualizationConfig from 'views/logic/aggregationbuilder/visualizations/LineVisualizationConfig';
+import AreaVisualizationConfig from 'views/logic/aggregationbuilder/visualizations/AreaVisualizationConfig';
+import type { InterpolationMode } from 'views/logic/aggregationbuilder/visualizations/Interpolation';
+
+import { Alert, Button, Row, Col } from 'components/graylog';
+import Spinner from 'components/common/Spinner';
+
+type LegacySeries = 'mean' | 'max' | 'min' | 'total' | 'count' | 'cardinality';
+type LegacyInterpolation = 'linear' | 'step-after' | 'basis' | 'bundle' | 'cardinal' | 'monotone';
+type LegacyInterval = 'minute' | 'hour' | 'day' | 'week' | 'month' | 'quarter' | 'year'
+type LegacyVisualization = 'bar' | 'area' | 'line' | 'scatterplot'
+type LegacyFieldChart = {
+  field: string,
+  renderer: LegacyVisualization,
+  interpolation: LegacyInterpolation,
+  valuetype: LegacySeries,
+  interval: LegacyInterval,
+}
+
+const Actions = styled.div`
+  margin-top: 10px;
+`;
+
+const mapSeries = (legacySeries: LegacySeries, field: string) => {
+  let seriesName;
+  switch (legacySeries) {
+    case 'total':
+      seriesName = 'sum';
+      break;
+    case 'mean':
+      seriesName = 'avg';
+      break;
+    case 'cardinality':
+      seriesName = 'card';
+      break;
+    default:
+      seriesName = legacySeries;
+  }
+  return `${seriesName}(${field})`;
+};
+
+const mapVisualization = (legacyVisualization: LegacyVisualization) => {
+  switch (legacyVisualization) {
+    case 'scatterplot':
+      return 'scatter';
+    default:
+      return legacyVisualization;
+  }
+};
+
+const mapTime = (legacyTime: string) => {
+  switch (legacyTime) {
+    case 'quarter':
+      return { unit: 'months', value: 3 };
+    default:
+      return { unit: `${legacyTime}s`, value: 1 };
+  }
+};
+
+const createVisualizationConfig = (legacyInterpolation: LegacyInterpolation, visualization: string) => {
+  let interpolation: InterpolationMode;
+  switch (legacyInterpolation) {
+    case 'basis':
+    case 'bundle':
+    case 'cardinal':
+    case 'monotone':
+      interpolation = 'spline';
+      break;
+    case 'linear':
+    case 'step-after':
+      interpolation = legacyInterpolation;
+      break;
+    default:
+      interpolation = 'linear';
+  }
+
+  switch (visualization) {
+    case 'line':
+      return new LineVisualizationConfig(interpolation);
+    case 'area':
+      return new AreaVisualizationConfig(interpolation);
+    default:
+      return undefined;
+  }
+};
+
+const _initialRowOffset = (widgetPositions) => {
+  const lastWidgetPosition = maxBy(values(widgetPositions), (position: WidgetPosition): number => position.row);
+  return lastWidgetPosition ? (lastWidgetPosition.row + lastWidgetPosition.height) : 0;
+};
+
+const _onMigrate = (legacyCharts: Array<LegacyFieldChart>, setMigrating: boolean => void, setMigrationFinished: boolean => void) => {
+  setMigrating(true);
```

Comment:
We could make the API simpler by just returning a `Promise` that wraps the migration code and is chained with `ViewStatesActions.update`, instead of passing `setMigrating`/`setMigrationFinished`.

The `onClick` handler would be separated from the actual migration code. It would call `setMigrating` when starting and `setMigrationFinished` after the promise was resolved, but the migration code itself would not have to take care of updating the UI.